iceoryx_examples/waitset/ice_waitset_trigger.cpp

```diff
@@ -97,56 +97,30 @@ class MyTriggerClass
         std::cout << "action performed" << std::endl;
     }
 
-
-    template <uint64_t>
-    friend class iox::popo::WaitSet;
+    friend iox::popo::EventAccessor;
 
   private:
+    /// @brief Only usable by the WaitSet, not for public use
     // This method attaches an event of the class to a waitset.
     // The event is choosen by the event parameter. Additionally, you can
     // set a eventId to group multiple instances and a custom callback.
-    iox::cxx::expected<iox::popo::WaitSetError>
-    enableEvent(iox::popo::WaitSet<>& waitset,
-                const MyTriggerClassEvents event,
-                const uint64_t eventId,
-                const iox::popo::Trigger::Callback<MyTriggerClass> callback) noexcept
+    void enableEvent(iox::popo::TriggerHandle&& triggerHandle, const MyTriggerClassEvents event) noexcept
     {
         switch (event)
         {
         case MyTriggerClassEvents::PERFORMED_ACTION:
-        {
-            return waitset
-                .acquireTriggerHandle(this,
-                                      // trigger calls this method to ask if it was triggered
-                                      {*this, &MyTriggerClass::hasPerformedAction},
-                                      // method which will be called when the waitset goes out of scope
-                                      {*this, &MyTriggerClass::disableEvent},
-                                      eventId,
-                                      callback)
-                // assigning the acquired trigger from the waitset to m_actionTrigger
-                .and_then([this](iox::popo::TriggerHandle& trigger) { m_actionTrigger = std::move(trigger); });
-        }
+            m_actionTrigger = std::move(triggerHandle);
+            break;
         case MyTriggerClassEvents::ACTIVATE:
-        {
-            return waitset
-                .acquireTriggerHandle(this,
-                                      // trigger calls this method to ask if it was triggered
-                                      {*this, &MyTriggerClass::isActivated},
-                                      // method which will be called when the waitset goes out of scope
-                                      {*this, &MyTriggerClass::disableEvent},
-                                      eventId,
-                                      callback)
-                // assigning the acquired trigger from the waitset to m_activateTrigger
-                .and_then([this](iox::popo::TriggerHandle& trigger) { m_activateTrigger = std::move(trigger); });
-        }
+            m_activateTrigger = std::move(triggerHandle);
```

Comment:
I totally misread this first and thought both cases would assign the handle to the same member, since they are only slightly different in the middle of the name. Could you give them more distinct names?